### ISSUE TITLE
Bats Making Raiders Permanantly Stuck

### ISSUE THREAD

**Issue by PoliteCanuck** [Scarabol/rock-raiders-web#7]
Im not sure if it's just random chance but if a Bat scares a Raider there's a chance to it make them become unresponsive. They just stand there and cannot be interacted with in any way anymore. This is most noticeable to me on the level Hot Stuff due to the increased number of bats.

**Comment by PoliteCanuck** [comment]
I've kept trying to beat Hot Stuff, but this glitch makes it frustratingly impossible. I've noticed that the glitch only happens when a rock raider is off screen. There was an incident with lava monsters that may have also triggered this. Though I can't confirm since it was off screen. It seems that any time Rock Raiders are frightened can trigger this, but due to cramp tunnels and lots of bats, this just causes the amount of times a Rock Raider to be scared to be much greater than other levels. 

I have occasionally seen it on other levels with bats, but only ever once per level and usually after I have a base set up so it's not an issue. But on Hot Stuff I've had two of my three Raiders get infected with this before my teleport pad is even built. It's the only level I find next to impossible to beat, since air is limited and a glitched Raider still consumes air.   

Is there any way to disable bats altogether? As infamous as they already are in Rock Raiders, this is the simplest fix I can think of and only someone wanting the purest experience will miss. Otherwise I greatly appreciate this project and have greatly enjoyed this so far!

**Comment by Scarabol** [comment]
Hi, thank you for the feedback and sorry to hear about this issue. I did some testing and could not find any issue with the bats itself.
![Rock Raiders Web 2024-08-23 06_27_23 944](https://github.com/user-attachments/assets/4d8749cb-f15a-4376-b145-deea163f6916)

Even three raiders stuck in a cave with a bat, never become unresponsive. Since the bats make the raider stop their jobs and drop stuff, I think it's related to some job/task they do while they get scared. Can you provide a screenshot of a stuck raider with "bubble" above their heads? Or did you notice this issue related to any specific job/task?

You can remove bats by modding the WAD files and removing them from the levels object list. Otherwise I'm working on a config dialog option, which could include an option to remove bats :wink:

**Comment by PoliteCanuck** [comment]
I actually saw this happen for the first time while replaying Hot Stuff, so I can confirm it is at the very least caused by when a Raider gets scared that triggers this.
![Rock Raiders Web 2024-08-23 20_51_40 570](https://github.com/user-attachments/assets/c3955446-d08d-43ec-9786-047105004804)

The scared Raider does not move and cannot be selected in anyway. I was trying to prevent lava erosion, so the Raider was probably trying to get ore at the tool store. Maybe if the Raider is trying to complete a task that keeps getting blocked by a bat it eventually gives up and breaks?

I am playing on a Mac just in case that does matter!

**Comment by PoliteCanuck** [comment]
Another instance of this happening on Ice Spy. I believe this Raider was carrying ore for a building.

![Uploading Rock Raiders Web 2024-08-24 05_29_41.952.png…]()

**Comment by PoliteCanuck** [comment]
Okay one last update. I noticed a Raider carrying ore near a wall that dynamite was being placed at. The Raider put down the ore, right as the dynamite started counting down and trigged the panic reaction. However instead the Raider just stood there and let themselves be blown up by the dynamite, even thought they had plenty of time to get away. Unfortunately I couldn't confirm because they died just as I realized what happened, but I think that might be what's specifically causing them to freeze. Most of the Raiders in Hot Stuff were carrying ore to stop the erosion, so it defiantly tracks with why they kept breaking.
